src/module/authentication.ts

```diff
@@ -83,6 +78,45 @@ class Authentication {
     // If the token is valid follow the next route.
     next();
   }
-}
 
-export default Authentication;
+  public static async grantDevAccess(req: Request, res: Response, next: NextFunction): Promise<void> {
+    const jwtToken: string = req.get('Authorization');
+
+    // Check if a authorization header is set.
+    if (jwtToken === undefined) {
+      new APIResponse(res, 401, {}, [
+        {
+          userMessage: 'Permission not granted.',
+          internalMessage: 'No JWT token was provided.',
+        },
+      ]).send();
+      return;
+    }
+
+    // If the header exists check the JWT token.
+    const validToken: JwtToken = await Authentication.verifyJwtToken(jwtToken);
+
+    try {
+      const userRepository = getRepository(User);
+      const user = await userRepository.findOneOrFail({
+        where: [{ login_name: validToken.username }],
+      });
+
+      if (validToken === null || !user.is_dev) {
+        new APIResponse(res, 401, {}, [
+          {
+            userMessage: 'Permission not granted.',
+            internalMessage: 'Wrong JWT token was provided or user was not a development account.',
+          },
+        ]).send();
+        return;
+      }
+    } catch (error) {
+      console.log('There is no user with username ' + validToken.username);
```

Comment:
also fails, when validToken === null, because the error in line 102 will cause an error and go into the catch before calling the return statement